In [2]:
import pandas as pd

In [5]:
df = pd.read_csv(r'D:\Purwadhika\belajar aja\learning\capstone_2\data\Airbnb Listings Bangkok.csv')
df.drop(['Unnamed: 0'], axis=1 ,inplace=True)
df.head()

id                                            name  host_id  \
0   27934                 Nice room with superb city view   120437   
1   27979                  Easy going landlord,easy place   120541   
2   28745               modern-style apartment in Bangkok   123784   
3   35780  Spacious one bedroom at The Kris Condo Bldg. 3   153730   
4  941865                      Suite Room 3 at MetroPoint   610315   

    host_name neighbourhood  latitude  longitude        room_type  price  \
0      Nuttee   Ratchathewi  13.75983  100.54134  Entire home/apt   1905   
1         Emy       Bang Na  13.66818  100.61674     Private room   1316   
2  Familyroom     Bang Kapi  13.75232  100.62402     Private room    800   
3     Sirilak     Din Daeng  13.78823  100.57256     Private room   1286   
4       Kasem     Bang Kapi  13.76872  100.63338     Private room   1905   

   minimum_nights  number_of_reviews last_review  reviews_per_month  \
0               3                 65  2020-01-06               0.50   
1               1                  0         NaN                NaN   
2              60                  0         NaN                NaN   
3               7                  2  2022-04-01               0.03   
4               1                  0         NaN                NaN   

   calculated_host_listings_count  availability_365  number_of_reviews_ltm  
0                               2               353                      0  
1                               2               358                      0  
2                               1               365                      0  
3                               1               323                      1  
4                               3               365                      0

# **DATA**
* id: Identitas unik Airbnb untuk tempat tersebut
* name: Nama daftar
* host_id:Identitas unik Airbnb untuk tuan rumah/pengguna
* neighborhood:Lingkungan yang diberi kode geografis menggunakan garis lintang dan bujur terhadap lingkungan yang ditentukan oleh shapefile digital terbuka atau publik
* latitude: Proyeksi Sistem Geodesi Dunia (WGS84) untuk garis lintang dan garis bujur.
* longitude:  Proyeksi Sistem Geodesi Dunia (WGS84) untuk garis lintang dan garis bujur.
* room_type:
* price: Harga harian dalam mata uang lokal. Catatan, tanda $ dapat digunakan terlepas dari lokasinya
* minimum_nights: jumlah minimum masa inap malam untuk tempat tersebut
* number_of_reviews: Jumlah ulasan yang dimiliki listingan
* last_review: Tanggal review terakhir/terbaru
* reviews_per_month:
* calculated_host_listings_count:
* availability_365:
* number_of_reviews_ltm:


In [6]:
df.shape

(15854, 16)

In [8]:
df.duplicated().sum() #menghitung duplikat pada seluruh data

0

Data tidak ada yang duplicate

In [9]:
missing_values = df.isnull().sum() #menghitung data yg null tiap kolom
missing_values

id                                   0
name                                 8
host_id                              0
host_name                            1
neighbourhood                        0
latitude                             0
longitude                            0
room_type                            0
price                                0
minimum_nights                       0
number_of_reviews                    0
last_review                       5790
reviews_per_month                 5790
calculated_host_listings_count       0
availability_365                     0
number_of_reviews_ltm                0
dtype: int64

In [19]:
# persentase missing value di tiap kolom
df.isna().sum()/df.shape[0]*100

id                                 0.000000
name                               0.050460
host_id                            0.000000
host_name                          0.006308
neighbourhood                      0.000000
latitude                           0.000000
longitude                          0.000000
room_type                          0.000000
price                              0.000000
minimum_nights                     0.000000
number_of_reviews                  0.000000
last_review                       36.520752
reviews_per_month                 36.520752
calculated_host_listings_count     0.000000
availability_365                   0.000000
number_of_reviews_ltm              0.000000
dtype: float64

Penjelasan
* *missing value* di kolom `name`, `host_name` memiliki presentase yang rendah (<10%), dengan missing value di kolom `last_review` dan `reviews_per_month` mencapai 36.5%


In [10]:
df[df['name'].isna()] #checking name yang berisi null

id name   host_id        host_name neighbourhood  latitude  \
439    4549768  NaN  18852579          Titawan  Phra Khanong  13.69406   
544    4720818  NaN  24386225           Cherry     Din Daeng  13.77562   
572    4245018  NaN  22030043        Parichart    Bang Phlat  13.78376   
669    6148415  NaN  31895202            Chira       Bang Na  13.68276   
1030   8055144  NaN  42521288          Nantida       Vadhana  13.74126   
1282  10000742  NaN  51374914  Diamond Bangkok   Ratchathewi  13.75328   
1594  10710165  NaN  55347997      Khaneungnit       Vadhana  13.71757   
2075  13142743  NaN  73275200         Pakaphol   Khlong Toei  13.72566   

      longitude        room_type  price  minimum_nights  number_of_reviews  \
439   100.59619     Private room   1080               5                  0   
544   100.57346     Private room   1200               1                  0   
572   100.49821     Private room   1200               1                  0   
669   100.60894  Entire home/apt   2424               2                  0   
1030  100.55761     Private room   5000               3                  0   
1282  100.52928     Private room    930               1                  6   
1594  100.60464     Private room   1000               1                  0   
2075  100.56416     Private room    850               1                  2   

     last_review  reviews_per_month  calculated_host_listings_count  \
439          NaN                NaN                               1   
544          NaN                NaN                               1   
572          NaN                NaN                               1   
669          NaN                NaN                               1   
1030         NaN                NaN                               1   
1282  2017-05-13               0.07                               1   
1594         NaN                NaN                               1   
2075  2017-12-11               0.03                               3   

      availability_365  number_of_reviews_ltm  
439                365                      0  
544                365                      0  
572                365                      0  
669                365                      0  
1030               365                      0  
1282               365                      0  
1594               365                      0  
2075               220                      0

In [11]:
df[df['host_name'].isna()]

id           name    host_id host_name neighbourhood  latitude  \
3571  19682464  Cozy Hideaway  137488762       NaN     Bang Kapi  13.76999   

      longitude     room_type  price  minimum_nights  number_of_reviews  \
3571  100.63769  Private room   1399               3                  1   

     last_review  reviews_per_month  calculated_host_listings_count  \
3571  2017-07-29               0.02                               1   

      availability_365  number_of_reviews_ltm  
3571               365                      0

In [21]:
df[df['number_of_reviews_ltm']!=0]

id                                               name  \
3                   35780     Spacious one bedroom at The Kris Condo Bldg. 3   
11                1765918                     2BR apt in a cozy neighborhood   
19                1793000                   The Duplex - Asoke- Luxury 92sqm   
28                 145343                Boutique Rooms Near Bangkok Airport   
30                 156583  Studio near Chula University/Silom walk to MRT...   
...                   ...                                                ...   
15712  785741287659406453                        ใจกลางเมืองติดห้างไอคอนสยาม   
15728  785976692600131294      1br/Free pool&gym/WIFI-Asok/SukhumvitBTS! 2PP   
15743  786248090308669514                 Vibrant Luxe 2 Bedroom | Thong Lor   
15744  786318268883527580                 Vibrant Luxe 2 Bedroom | Thong Lor   
15796  788841933134248110  Stunning river view in the heart of BKK 5min/t...   

         host_id host_name neighbourhood   latitude   longitude  \
3         153730   Sirilak     Din Daeng  13.788230  100.572560   
11       9279712      Jing    Phaya Thai  13.774860  100.542720   
19       9407280      Timo       Vadhana  13.746680  100.561370   
28        703944   Parinya   Lat Krabang  13.721868  100.771713   
30         58920      Gael      Bang Rak  13.728500  100.523130   
...          ...       ...           ...        ...         ...   
15712  200814460       Noi     Thon buri  13.696506  100.486226   
15728  485536928     Lucas   Khlong Toei  13.734856  100.557960   
15743   46163812    Ernest       Vadhana  13.730126  100.586369   
15744   46163812    Ernest       Vadhana  13.729880  100.586269   
15796  315867023      Alex      Bang Rak  13.719792  100.515910   

             room_type  price  minimum_nights  number_of_reviews last_review  \
3         Private room   1286               7                  2  2022-04-01   
11     Entire home/apt   1893              15                129  2022-09-30   
19     Entire home/apt   5034              21                287  2022-11-22   
28        Private room   1329               1                 28  2022-11-25   
30     Entire home/apt   1176               7                 63  2022-11-25   
...                ...    ...             ...                ...         ...   
15712  Entire home/apt   2000               1                  1  2022-12-25   
15728  Entire home/apt   2514               1                  1  2022-12-26   
15743  Entire home/apt   3932               1                  3  2022-12-24   
15744  Entire home/apt   4285              28                  3  2022-12-28   
15796  Entire home/apt   3304               2                  2  2022-12-28   

       reviews_per_month  calculated_host_listings_count  availability_365  \
3                   0.03                               1               323   
11                  1.17                               1               356   
19                  2.59                               1               349   
28                  0.28                               1               349   
30                  0.47                               2                95   
...                  ...                             ...               ...   
15712               1.00                               2               361   
15728               1.00                               4               257   
15743               3.00                               8               349   
15744               3.00                               8               365   
15796               2.00                               3               342   

       number_of_reviews_ltm  
3                          1  
11                         1  
19                         3  
28                        13  
30                         2  
...                      ...  
15712                      1  
15728                      1  
15743                      3  
15744                      3  
15796             

In [17]:
df_null_review = df[df['number_of_reviews']==0]

In [18]:
df_null_review.isnull().sum()

id                                   0
name                                 6
host_id                              0
host_name                            0
neighbourhood                        0
latitude                             0
longitude                            0
room_type                            0
price                                0
minimum_nights                       0
number_of_reviews                    0
last_review                       5790
reviews_per_month                 5790
calculated_host_listings_count       0
availability_365                     0
number_of_reviews_ltm                0
dtype: int64

In [ ]:
df_cleaned_column = df.dropna(subset=['last_review', 'reviews_per_month'])
df_cleaned_column

id                                               name  \
0                   27934                    Nice room with superb city view   
3                   35780     Spacious one bedroom at The Kris Condo Bldg. 3   
5                 1704776                     NEw Pro!!  Bungalow Bkk Centre   
6                   48736                    Condo with Chaopraya River View   
8                 1744248                 Batcave, Pool view, near Chatuchak   
...                   ...                                                ...   
15712  785741287659406453                        ใจกลางเมืองติดห้างไอคอนสยาม   
15728  785976692600131294      1br/Free pool&gym/WIFI-Asok/SukhumvitBTS! 2PP   
15743  786248090308669514                 Vibrant Luxe 2 Bedroom | Thong Lor   
15744  786318268883527580                 Vibrant Luxe 2 Bedroom | Thong Lor   
15796  788841933134248110  Stunning river view in the heart of BKK 5min/t...   

         host_id host_name  neighbourhood   latitude   longitude  \
0         120437    Nuttee    Ratchathewi  13.759830  100.541340   
3         153730   Sirilak      Din Daeng  13.788230  100.572560   
5        2129668  Wimonpak  Bang Kho laen  13.697570  100.528800   
6         222005  Athitaya     Rat Burana  13.685560  100.495350   
8        9181769       Nol     Chatu Chak  13.816930  100.564330   
...          ...       ...            ...        ...         ...   
15712  200814460       Noi      Thon buri  13.696506  100.486226   
15728  485536928     Lucas    Khlong Toei  13.734856  100.557960   
15743   46163812    Ernest        Vadhana  13.730126  100.586369   
15744   46163812    Ernest        Vadhana  13.729880  100.586269   
15796  315867023      Alex       Bang Rak  13.719792  100.515910   

             room_type  price  minimum_nights  number_of_reviews last_review  \
0      Entire home/apt   1905               3                 65  2020-01-06   
3         Private room   1286               7                  2  2022-04-01   
5      Entire home/apt   1000             250                 19  2017-08-03   
6         Private room   1558               3                  1  2014-02-03   
8      Entire home/apt   1558               3                 10  2016-03-29   
...                ...    ...             ...                ...         ...   
15712  Entire home/apt   2000               1                  1  2022-12-25   
15728  Entire home/apt   2514               1                  1  2022-12-26   
15743  Entire home/apt   3932               1                  3  2022-12-24   
15744  Entire home/apt   4285              28                  3  2022-12-28   
15796  Entire home/apt   3304               2                  2  2022-12-28   

       reviews_per_month  calculated_host_listings_count  availability_365  \
0                   0.50                               2               353   
3                   0.03                               1               323   
5                   0.17                              41               365   
6                   0.01                               1               365   
8                   0.09                               1               365   
...                  ...                             ...               ...   
15712               1.00                               2               361   
15728               1.00                               4               257   
15743               3.00                               8               349   
15744               3.00                               8               365   
15796               2.00                               3               342   

       number_of_reviews_ltm  
0                          0  
3                          1  
5                          0  
6                          0  
8                          0  
...                      ...  
15712                      1  
15728                      1  
15743                      3  
15744                      3  
15796 

In [ ]:
df_cleaned_column.isnull().sum()

id                                0
name                              2
host_id                           0
host_name                         0
neighbourhood                     0
latitude                          0
longitude                         0
room_type                         0
price                             0
minimum_nights                    0
number_of_reviews                 0
last_review                       0
reviews_per_month                 0
calculated_host_listings_count    0
availability_365                  0
number_of_reviews_ltm             0
dtype: int64

In [ ]:
df_cleaned_column.name.nunique()

9448

In [ ]:
# pokoknye itu data sebisa mungkin jangan diapus, kecuali <10% jumlah yg missing_values

# 2. buat ganti isi yg missing > perlu dianalisa, apakah harus diganti apa tetep dibiarkan.